redbot/cogs/general/general.py

```diff
@@ -202,54 +202,71 @@ def __init__(self):
+        joined = user.joined_at
+        userstatus = user.status
+        playinggame = user.game
+        userisbot = user.bot
+        roles = [x.name for x in user.roles if x.name != "@everyone"]
         if roles:
-            roles = ", ".join([x.name for x in roles])
+            roles = sorted(roles, key=[x.name for x in server.role_hierarchy
+                                       if x.name != "@everyone"].index)
+            roles = ", ".join(roles)
         else:
-            roles = _("None")
-
-        data = discord.Embed(description=game, colour=user.colour)
-        data.add_field(name=_("Joined Discord on"), value=created_on)
-        data.add_field(name=_("Joined this guild on"), value=joined_on)
-        data.add_field(name=_("Roles"), value=roles, inline=False)
-        data.set_footer(text=_("Member #{} | User ID: {}"
-                               "").format(member_number, user.id))
-
-        name = str(user)
-        name = " ~ ".join((name, user.nick)) if user.nick else name
-
-        if user.avatar:
-            data.set_author(name=name, url=user.avatar_url)
-            data.set_thumbnail(url=user.avatar_url)
+            roles = "None"
+        if user.bot is False:
+            userisbot = "Requested user is not a bot"
         else:
-            data.set_author(name=name)
-
-        try:
-            await ctx.send(embed=data)
+            userisbot = "Requested user is a bot"
+        if user.nick is None:
+            displayname = "User currently has no nickname"
+        else:
+            displayname = user.nick
+        if user.top_role is "@everyone":
+            highestrole = "User does not have any roles"
+        else:
+            highestrole = user.top_role
+        if user.game is None:
+            displayname = "User currently has no nickname"
+        else:
+            displayname = user.nick
+        color = ''.join([choice('0123456789ABCDEF') for x in range(6)])
+        color = int(color, 16)
+        embed=discord.Embed(title="Username", description=realname, color=color)
+        embed.set_thumbnail(url=str(useravatarurl))
+        embed.add_field(name="User's ID", value=userid, inline=False)
+        embed.add_field(name="Highest Role", value=highestrole, inline=False)
+        embed.add_field(name="All user roles", value=roles, inline=False)
+        embed.add_field(name="Current Nickname", value=displayname, inline=False)
+        embed.add_field(name="Status", value=userstatus, inline=False)
+        embed.add_field(name="User is a bot", value=userisbot, inline=False)
+        embed.add_field(name="Playing", value=playinggame, inline=False)
+        embed.add_field(name="Joined Discord on", value=created_on, inline=False)
+        embed.add_field(name="Joined this guild on", value=joined_on, inline=True)
+        embed.set_footer(text="Member #{} | User ID:{}"
+                             "".format(member_number, user.id))
+        try :
+            await self.bot.say(embed=embed)
```

Comment:
Entire embed is missing ``_(...)`` formatting. A special case for the translation backbone.

Reply:
and `self.bot.say` does not exist in v3. This should be `ctx.send` instead